harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -38,6 +38,12 @@ def __init__(self, trial_inst: det.Trial, *args: Any, **kwargs: Any) -> None:
         if torch.cuda.is_available():
             self.prof._set_sync_device(self._sync_device)
         self.callbacks = self.trial.build_callbacks()
+        for callback in self.callbacks.values():
+            if util.is_overridden(callback.on_checkpoint_end, pytorch.PyTorchCallback):
+                logging.warning(
+                    "The on_checkpoint_end callback is deprecated, please use on"
+                    "checkpoint_write_end instead"
+                )
```

Comment:
can you use a FutureWarning, which automatically guarantees that the line only prints once?

There's a good example of a FutureWarning [here](https://github.com/determined-ai/determined/blob/master/harness/determined/common/experimental/checkpoint/_checkpoint.py#L216-L224).

Reply:
Done.